### PR TITLE
Fixed issue with false positives of `0` & `1` boolean values and circleCI job failure fix

### PR DESCRIPTION
**Pull request by vishalp-dev**
transform

# Description of change
- Modified the `_transform` method of the Transformer class.
- Formatting and indentation 
- Added chunk decode logic to make code Python version compatible

# Manual QA steps
 - 
 
# Risks
 - 
 
# Rollback steps
 - revert this branch

co-author @rdeshmukh15

### CODE REVIEW

File: tap_workday_raas/discover.py [hunk 1]
```diff
@@ -87,45 +80,44 @@ def generate_schema_for_report(xsd):
...
         streams.append(
             {
-                'stream': report['report_name'],
-                'tap_stream_id': report['report_name'],
-                'schema': schema,
-                'metadata': stream_md
+                "stream": report["report_name"],
+                "tap_stream_id": report["report_name"],
+                "schema": schema,
+                "metadata": stream_md,
```
Comment (dbshah1212) [inline tap_workday_raas/discover.py]: Comma at the end of dict not required.

File: tap_workday_raas/discover.py [hunk 2]
```diff
@@ -87,45 +80,44 @@ def generate_schema_for_report(xsd):
...
 
-        xsd = download_xsd(report['report_url'], username, password)
+        xsd = download_xsd(report["report_url"], username, password)
         schema = generate_schema_for_report(xsd)
 
         stream_md = metadata.get_standard_metadata(schema,
-                                                   key_properties=report.get('key_properties'),
-                                                   replication_method='FULL_TABLE')
+                                                   key_properties=report.get("key_properties"),
+                                                   replication_method="FULL_TABLE",)
```
Comment (dbshah1212) [inline tap_workday_raas/discover.py]: Comma at the end not required.

File: tap_workday_raas/discover.py [hunk 3]
```diff
@@ -87,45 +80,44 @@ def generate_schema_for_report(xsd):
...
                 raise Exception("Found unexpected value for maxOccurs attribute: '{}'".format(max_occurs))
 
-            is_list = max_occurs == 'unbounded'
+            is_list = max_occurs == "unbounded"
 
             if is_list:
-                elem_schema = {'type': 'array',
-                               'items': complex_type_mapping[elem_type]}
+                elem_schema = {"type": "array",
+                               "items": complex_type_mapping[elem_type],}
```
Comment (dbshah1212) [inline tap_workday_raas/discover.py]: Comma at the end of dict not required.

File: tap_workday_raas/transform.py [hunk 4]
```diff
@@ -0,0 +1,20 @@
...
+
+class WorkdayTransformer(Transformer):
+    """
+        Modified Transformer to identify `0` and `1` as boolean values
+    """
+    def _transform(self, data, typ, schema, path):
+        if self.pre_hook:
+            data = self.pre_hook(data, typ, schema)
+        if typ == "boolean":
+            if isinstance(data, str) and (data.lower() == "false" or data.lower() == "0"):
```
Comment (dbshah1212) [inline tap_workday_raas/transform.py]:
```suggestion
            if isinstance(data, str) and (data.lower() in ("false","0")):
```

File: tap_workday_raas/discover.py (hunk 1 above)
Comment (rdeshmukh15) [reply tap_workday_raas/discover.py]: done

File: tap_workday_raas/discover.py (hunk 2 above)
Comment (rdeshmukh15) [reply tap_workday_raas/discover.py]: done

File: tap_workday_raas/discover.py (hunk 3 above)
Comment (rdeshmukh15) [reply tap_workday_raas/discover.py]: done

File: tap_workday_raas/transform.py (hunk 4 above)
Comment (rdeshmukh15) [reply tap_workday_raas/transform.py]: done

**Review by dbshah1212 — APPROVED** [review]

**Review by dbshah1212 — APPROVED** [review]

**Review by savan-chovatiya — CHANGES_REQUESTED** [review]
Can you please add unit test cases for different scenarios of boolean values?

**Review by savan-chovatiya — APPROVED** [review]

**Pull request merged**